React/Views/RCTShadowView.m

```diff
@@ -263,6 +263,20 @@ - (CGRect)measureLayoutRelativeToAncestor:(RCTShadowView *)ancestor
   return (CGRect){offset, self.frame.size};
 }
 
+- (BOOL)viewIsAncestorOf:(RCTShadowView *)ancestor
+{
+  NSInteger depth = 30; // max depth to search
```

Comment:
I don't like that either and I would love to know why the magic number is 30 and not 20 or 45 😄. But in order to keep consistency, I decided to keep that untouched.